frontend/e2e/support/backend_server.py

```
@@ -0,0 +1,54 @@
+import uvicorn
+
+sys.path.insert(0, str(Path(__file__).resolve().parents[3]))
+app_module: Any = importlib.import_module("backend.app")
+api_proxy_module: Any = importlib.import_module("backend.api_proxy")
+
+
+async def _public_ip(*_args: Any, **_kwargs: Any) -> str:
+    return "192.0.2.100"
+
+
+async def _ipinfo(*_args: Any, **_kwargs: Any) -> dict[str, str]:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)